lib/libavc.py

```diff
@@ -11,58 +11,65 @@
     self.name = name
     self.codecname = 'libavc'
     super(LibavcCodec, self).__init__(
-      name,
-      formatter=(formatter or encoder.OptionFormatter(prefix='--', infix=' ')))
+        name,
+        formatter=(formatter or
+                   encoder.OptionFormatter(prefix='--', infix=' ')))
     self.extension = 'avi'
     self.option_set = encoder.OptionSet(
-      # Rate control. 0 = constant QP, 1 = storage, 2 = CBR high delay,
-      # 3 = CBR low delay
-      # 2 and 3 seems to drop frames - sometimes, but not always.
-      # 3 is able to run out of memory.
-      encoder.Option('rc', ['0', '1']),
+        # Rate control. 0 = constant QP, 1 = storage, 2 = CBR high delay,
+        # 3 = CBR low delay
+        # 2 and 3 seems to drop frames - sometimes, but not always.
+        # 3 is able to run out of memory.
+        encoder.Option('rc', ['0', '1']),
     )
 
   def StartEncoder(self, context):
     return encoder.Encoder(context, encoder.OptionValueSet(self.option_set, ''))
 
   def EncodeCommandLine(self, parameters, bitrate, videofile, encodedfile):
     commandline = (
-      '%(tool)s '
-      '--width %(width)d --height %(height)d '
-      '--src_framerate %(framerate)d --tgt_framerate %(framerate)d '
-      '--input_chroma_format YUV_420P --bitrate %(bitrate)d '
-      '%(parameters)s '
-      '--input %(inputfile)s --output %(outputfile)s' % {
-        'tool': encoder.Tool('avcenc'),
-        'width': videofile.width,
-        'height': videofile.height,
-        'framerate': videofile.framerate,
-        'bitrate': bitrate * 1000, # Bitrate is in bits/sec, not kilobits.
-        'parameters': parameters.ToString(),
-        'inputfile': videofile.filename,
-        'outputfile': encodedfile}
+        '%(tool)s '
+        '--width %(width)d --height %(height)d '
+        '--src_framerate %(framerate)d --tgt_framerate %(framerate)d '
+        '--input_chroma_format YUV_420P --bitrate %(bitrate)d '
+        '%(parameters)s '
+        '--input %(inputfile)s --output %(outputfile)s' % {
+            'tool': encoder.Tool('avcenc'),
+            'width': videofile.width,
+            'height': videofile.height,
+            'framerate': videofile.framerate,
+            'bitrate': bitrate * 1000, # Bitrate is in bits/sec, not kilobits.
+            'parameters': parameters.ToString(),
+            'inputfile': videofile.filename,
+            'outputfile': encodedfile}
     )
     return commandline
 
   def DecodeCommandLine(self, videofile, encodedfile, yuvfile):
     commandline = "%s -loglevel warning -codec:v h264 -i %s %s" % (
-      encoder.Tool('ffmpeg'),
-      encodedfile, yuvfile)
+        encoder.Tool('ffmpeg'),
+        encodedfile, yuvfile)
     return commandline
 
   def ResultData(self, encodedfile):
     return {'frame': file_codec.FfmpegFrameInfo(encodedfile)}
+
+  def EncoderVersion(self):
+    # libavc doesn't appear to have a built-in version string. Use the
+    # git checksum instead.
```

Comment:
Worth marking the version with something like git# to state that the version is derived from a git hash? Also, worth splicing in a date, too? That might even create a meaningful ordering of version strings.